samples/bluetooth/bap_broadcast_sink/src/main.c

```diff
@@ -1033,15 +1019,31 @@ static int bis_sync_req_cb(struct bt_conn *conn,
 			   const struct bt_bap_scan_delegator_recv_state *recv_state,
 			   const uint32_t bis_sync_req[CONFIG_BT_BAP_BASS_MAX_SUBGROUPS])
 {
-	printk("BIS sync request received for %p: 0x%08x->0x%08x, broadcast id: 0x%06x, (%s)\n",
-	       recv_state, requested_bis_sync, bis_sync_req[0], recv_state->broadcast_id,
+	requested_bis_sync = 0;
+	requested_subgroup_sync = 0;
+
+	for (uint8_t subgroup = 0; subgroup < recv_state->num_subgroups; subgroup++) {
+		if (bis_sync_req[subgroup] != 0) {
+			/* bis_sync_req[subgroup] = 0b1 or BT_BAP_BIS_SYNC_NO_PREF */
+			if (requested_bis_sync == 0) {
+				requested_bis_sync = bis_sync_req[subgroup];
+			} else {
+				/* requested_bis_sync = 0b1 or BT_BAP_BIS_SYNC_NO_PREF */
+				if (requested_bis_sync != BT_BAP_BIS_SYNC_NO_PREF &&
+				    bis_sync_req[subgroup] != BT_BAP_BIS_SYNC_NO_PREF) {
+					printk("Bis sync request from more than 1 subgroup!\n");
+					return -EINVAL;
+				}
```

Comment:
Suggest to add a comment why we are rejecting this. Why are we? Doesn't this prevent us from syncing to multiple subgroups?

Reply:
> request from more than 1 subgroup is not possible unless sync value is 0 or BT_BAP_BIS_SYNC_NO_PREF.

Can you elaborate?
From a spec perspektive it's possible to sync to BIS 1 from Subgroup 1 and BIS 2 from Subgroup 2.

Reply:
> > request from more than 1 subgroup is not possible unless sync value is 0 or BT_BAP_BIS_SYNC_NO_PREF.
> 
> Can you elaborate? From a spec perspektive it's possible to sync to BIS 1 from Subgroup 1 and BIS 2 from Subgroup 2.

@Thalley  - would this make much sense in practice? (multi subgroup sync)

Reply:
I don't have a good use case for it to be honest :) But it's incorrect to say that it's not possible, and I wouldn't be against not supporting it in the sample, and if we do not support syncing to multiple subgroups then we remove `requested_subgroup_sync` again as well as add a comment why this isn't implemented.

Reply:
When reading "6.5.4. Adding broadcast sources" from BAP spec
 
"The Broadcast Assistant shall not write a value of 0b1 to any BIS_Sync parameter BIS_index value for more than one subgroup except if the Broadcast Assistant writes a value of 0xFFFFFFFF (no preference) to the BIS_Sync parameter for each of those subgroups."

You can be i doubt about sync request from multiple subgroups is possible. I prefer to leave it as is an make a comment.

Also it would be a strange use case to stream from multiple subgroups  at the same time :-)